File: moderate/decimal_to_binary/decimal.to.binary.cpp
// Reuben Thorpe (2016), CodeEval [Decimal To Binary v1.0]
#include <iostream>
#include <fstream>
#include <iomanip>
#include <bitset>

using namespace std;

int main(int argc, char* argv[]) {
  ifstream inFile(argv[1]);
  int num;
  cout << fixed << setprecision(0);

  while (inFile >> num) {
    cout << stod(bitset<32>(num).to_string()) << '\n';
  }

  return(0);
}

File: moderate/decimal_to_binary/decimal.to.binary.java
// Reuben Thorpe (2016), CodeEval [Decimal To Binary v1.1]
import java.io.*;
import java.util.Scanner;

public class Main {
  public static void main (String []args) throws IOException {
    File file = new File(args[0]);
    Scanner s = new Scanner(file);
    while (s.hasNext()){
      System.out.println(Integer.toBinaryString(s.nextInt()));
    }
  }
}
